packages/lockfile-file/src/read.ts

```diff
@@ -225,10 +226,15 @@ async function _readGitBranchLockfiles (
  * Reverts changes from the "forceSharedFormat" write option if necessary.
  */
 function convertFromLockfileFileMutable (lockfileFile: LockfileFile): Lockfile {
-  if (typeof lockfileFile?.['specifiers'] !== 'undefined') {
+  const specifiersFieldIsOnRoot = typeof lockfileFile?.['specifiers'] !== 'undefined'
+  // When inlineSpecifiersFormat is true, the specifiers field may not exist.
+  // Check for one of the dep type fields that were moved to the root instead.
+  const someDepFieldIsOnRoot = DEPENDENCIES_FIELDS.some(depType => lockfileFile[depType] != null)
+
+  if (specifiersFieldIsOnRoot || someDepFieldIsOnRoot) {
```

Comment:
When the shared lockfile format is used, `specifiers`, `dependencies`, `devDependencies`, and `optionalDependencies` may appear as keys on the root of the lockfile. At them moment this logic only checks for the `specifiers` key to guess whether the lockfile was serialized using the shared lockfile format.

After the changes in this PR, we unfortunately can't rely on the presence of `specifiers` on the root to detect the shared lockfile format since the "inline specifiers" format removes `specifiers`. We're now checking for one of the other fields. **This patch is only change not gated behind the feature flag.** Assuming the logic is correct, it should not change runtime behavior when the flag is off though.

If that's not okay, I can revert the change to this function and have `convertToInlineSpecifiersFormat` write out an empty `specifiers: {}` block. The thinking was that we'll have to remove the check for `specifiers` here if we end up adopting the changes in this feature flag, so I might as well commit this planned change and have it up for review.

Reply:
can't you just check the existence of "importers" field instead?

Reply:
Yes, that seems much better. Thanks for the suggestion.